libs/community/langchain_community/chat_models/tongyi.py

```diff
@@ -256,11 +320,57 @@ def stream_completion_with_retry(self, **kwargs: Any) -> Any:
         @retry_decorator
         def _stream_completion_with_retry(**_kwargs: Any) -> Any:
             responses = self.client.call(**_kwargs)
+            prev_resp = None
+
             for resp in responses:
-                yield check_response(resp)
+                # If we are streaming without `incremental_output = True`,
+                # we need to calculate the delta response manually
+                if _kwargs.get("stream") and not _kwargs.get(
+                    "incremental_output", False
+                ):
+                    if prev_resp is None:
+                        delta_resp = resp
+                    else:
+                        delta_resp = self.subtract_client_response(resp, prev_resp)
+                    prev_resp = resp
+                    yield check_response(delta_resp)
```

Comment:
@cheese-git would this represent a breaking change for streaming generally?

Reply:
No, it would not.

Before this PR, `incremental_output` is always `True` when streaming:

https://github.com/langchain-ai/langchain/blob/64c47224a0d91b06e0ed04a635c03db4c6760b71/libs/community/langchain_community/chat_models/tongyi.py#L416-L417


To support tools when streaming, I updated the condition to [this](https://github.com/cheese-git/langchain/blob/336f3b852237119c9591e762535bc37095f1d949/libs/community/langchain_community/chat_models/tongyi.py#L326-L330):

```python
# According to the Tongyi official docs,
# `incremental_output` with `tools` is not supported yet
if params.get("stream") and not params.get("tools"):
    params["incremental_output"] = True
```

So "streaming without `incremental_output` being `True`" is a new circumstance.